### DISCUSSION TITLE
built-in kafka support

### DISCUSSION THREAD

**Discussion by aryaniyaps** [django/channels#1654] [Ideas] [answered]
I was thinking if channels could support kafka built-in. (The kafka-python client installation is optional, but we'll warn if any KafkaWebsocketConsumer is used without the package). This way, projects can start using kafka immediately and will have horizontal scalability. I was thinking that the kafka consumers and producers could be modified by these settings.

```py
# settings.py
CHANNELS_KAFKA = {
    'cluster': 'localhost:4327',
    'consumers': {},
}
```
```py
class KafkaWebsocketConsumer(WebsocketConsumer):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        try:
           # kafka support is optional
           import kafka-python
        except ImportError:
            raise AssertionError(
                'kafka-python is not installed.'
                'kafka websocket consumers will not be supported.'
            )
```
this way, projects can choose between the groups API or kafka as the message distribution system.
I am willing to do every bit of this integration from consumers, updating the documentation and everything.

I just wanted to ask, is this worth doing?

**Comment by carltongibson** [comment, marked as answer]
Happy to link to a third party package, but not a consumer I'd want to support in channels itself.

**Reply by aryaniyaps** [reply]
thanks for the quick reply, I might work on this later, thanks!
